src/main/kotlin/de/bund/digitalservice/useid/tracking/matomo/MatomoTrackingService.kt

```diff
@@ -31,7 +31,9 @@ class MatomoTrackingService(trackingProperties: TrackingProperties, private val
     fun constructEventURL(e: MatomoEvent): String {
         val session = e.sessionId?.let { "&uid=$it" } ?: ""
         val userAgent = e.userAgent?.let { "&ua=${UriUtils.encode(e.userAgent, UTF_8)}" } ?: ""
-        val url = "$domain?idsite=$siteId&rec=1&ca=1&e_c=${e.category}&e_a=${e.action}&e_n=${e.name}$session$userAgent"
+        // custom dimension for tenantId -> https://matomo.org/faq/reporting-tools/create-track-and-manage-custom-dimensions/
+        val tenantId = e.tenantId?.let { "&dimension3=${UriUtils.encode(e.tenantId, UTF_8)}" } ?: ""
```

Comment:
I think dimension3 must be dynamic, on matomo production (site **useid** it is dimension1)

Reply:
Thanks for pointing that out 👍
I will think about a good solution!